cpp/src/arrow/json/parser.cc

```diff
@@ -17,28 +17,33 @@
 
 #include "arrow/json/parser.h"
 
-#include <algorithm>
-#include <cstdio>
 #include <limits>
-#include <sstream>
 #include <tuple>
+#include <unordered_map>
 #include <utility>
 #include <vector>
 
+#include "arrow/util/sse-util.h"
+#if defined(ARROW_HAVE_SSE4_2)
+#define RAPIDJSON_SSE42 1
```

Comment:
I think it would be worthwhile consolidating the RapidJson options and definitions somewhere. There's already some of them in `ipc/json-internal.h` but they could all go into a dedicated `.h` file (for example `json/rapidjson-defs.h`).